File: Domain/LeaveStatusHistory.cs
namespace Domain;

public class LeaveStatusHistory
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string AnnualLeaveId { get; set; } = string.Empty;
    public AnnualLeave? AnnualLeave { get; set; }
    public string ChangedByUserId { get; set; } = string.Empty;
    public User? ChangedByUser { get; set; }
    public AnnualLeaveStatus? OldStatus { get; set; }
    public AnnualLeaveStatus NewStatus { get; set; }
    public string? Comment { get; set; }
    public DateTime ChangedAt { get; set; } = DateTime.UtcNow;
}

File: client/src/lib/types/leave-status-history.ts
export interface LeaveStatusHistory {
    id: string
    annualLeaveId: string
    employeeId: string
    employeeName: string
    changedByUserId: string
    changedByUserName: string
    leaveTypeName: string | null
    oldStatus: string | null
    newStatus: string
    comment: string | null
    changedAt: string
}
